src/emulator/storage/apis/gcloud.ts

```diff
@@ -83,30 +83,35 @@ export function createCloudEndpoints(emulator: StorageEmulator): Router {
     return res.json(new CloudStorageObjectMetadata(updatedMetadata));
   });
 
-  gcloudStorageAPI.get("/b/:bucketId/o", (req, res) => {
+  gcloudStorageAPI.get("/b/:bucketId/o", async (req, res) => {
+    let listResponse: ListObjectsResponse;
     // TODO validate that all query params are single strings and are not repeated.
-    let maxRes = undefined;
-    if (req.query.maxResults) {
-      maxRes = +req.query.maxResults.toString();
+    try {
+      listResponse = await adminStorageLayer.listObjects({
+        bucketId: req.params.bucketId,
+        prefix: req.query.prefix ? req.query.prefix.toString() : "",
+        delimiter: req.query.delimiter ? req.query.delimiter.toString() : "",
+        pageToken: req.query.pageToken ? req.query.pageToken.toString() : undefined,
+        maxResults: req.query.maxResults ? +req.query.maxResults.toString() : undefined,
+        authorization: req.header("authorization"),
+      });
+    } catch (err) {
+      if (err instanceof ForbiddenError) {
+        return res.sendStatus(403);
```

Comment:
Not specific to this PR, but do you know why we aren't sending error messages in `gcloud` [like we do in `firebase.ts`](https://github.com/firebase/firebase-tools/blob/cdca8f24230987e6d1d54d2772aefce47e9a2350/src/emulator/storage/apis/firebase.ts#L165)

Seems like it would be useful in general, but perhaps it's because we don't have any guarantees of what GCS actually returns as the message besides the 403?

Reply:
https://cloud.google.com/storage/docs/json_api/v1/status-codes#403-forbidden

Seems it can be a plethora of messages, fine with leaving the 403 as is. I'll leave the comment open in case you have any thoughts, but feel free to resolve.

Reply:
Agreed, mainly just asked out of curiosity this shouldn't be our priority now. Thanks for the clarification!